ban-cooking.lua

```diff
@@ -0,0 +1,440 @@
+                    if s.value == "CHEESE_MAT" then
+                        local matinfo = dfhack.matinfo.find(c.creature_id, m.id)
+                        ban_cooking(c.name[2] .. ' ' .. m.id, matinfo.type, matinfo.index, df.item_type.LIQUID_MISC, -1)
+                        break
+                    end
+                end
+            end
+        end
+    end
+end
+
+funcs["oil"] = function()
+    for _, p in ipairs(df.global.world.raws.plants.all) do
+        for _, m in ipairs(p.material) do
+            if m.flags.EDIBLE_COOKED then
+                for _, s in ipairs(m.reaction_product.id) do
+                    if s.value == "SOAP_MAT" then
+                        local matinfo = dfhack.matinfo.find(p.id, m.id)
+                        ban_cooking(p.name .. ' ' .. m.id, matinfo.type, matinfo.index, df.item_type.LIQUID_MISC, -1)
+                        break
+                    end
+                end
+            end
+        end
+    end
+end
+
+funcs["seeds"] = function()
+    for _, p in ipairs(df.global.world.raws.plants.all) do
+        if p.material_defs.type.seed ~= -1 and p.material_defs.idx.seed ~= -1 and not p.flags.TREE then
+            ban_cooking(p.name + ' seeds', p.material_defs.type.seed, p.material_defs.idx.seed, df.item_type.SEEDS, -1)
+            for _, m in ipairs(p.material) do
+                if m.id == "STRUCTURAL" and m.flags.EDIBLE_COOKED then
+                    local has_drink = false
+                    local has_seed = false
+                    for _, s in ipairs(m.reaction_product.id) do
+                        has_seed = has_seed or s.value == "SEED_MAT"
+                        has_drink = has_drink or s.value == "DRINK_MAT"
+                    end
+                    if has_seed and has_drink then
+                        local matinfo = dfhack.matinfo.find(p.id, m.id)
+                        ban_cooking(p.name .. ' ' .. m.id, matinfo.type, matinfo.index, df.item_type.PLANT, -1)
+                    end
+                end
+            end
+            for k, g in ipairs(p.growths) do
+                local matinfo = dfhack.matinfo.decode(g)
+                local m = matinfo.material
+                if m.flags.EDIBLE_COOKED then
+                    local has_drink = false
+                    local has_seed = false
+                    for _, s in ipairs(m.reaction_product.id) do
+                        has_seed = has_seed or s.value == "SEED_MAT"
+                        has_drink = has_drink or s.value == "DRINK_MAT"
+                    end
+                    if has_seed and has_drink then
+                        ban_cooking(p.name .. ' ' .. m.id, matinfo.type, matinfo.index, df.item_type.PLANT_GROWTH, k)
+                    end
+                end
+            end
+        end
+    end
+end
+
+funcs["brew"] = function()
+    for _, p in ipairs(df.global.world.raws.plants.all) do
+        if p.material_defs.type.drink ~= -1 and p.material_defs.idx.drink ~= -1 then
```

Comment:
same here, invert the condition and use `goto` to remove an unneeded level of indentation